### PR TITLE
feat(ibis_bigquery/registry.py): add wiring for bigquery-supported ibis geospatial operations

### LINKED ISSUE #79: FEAT: Geospatial functions for BigQuery
BigQuery supports [geospatial features](https://cloud.google.com/bigquery/docs/gis-intro). We should look into adding support for this to ibis.

cc @tswast

### PR DESCRIPTION
**Pull request by santind**
fixes #79

### CODE REVIEW

**Comment by santind** [comment]
@tswast, do you need anything else from me before kicking off the test workflow for this PR? I'm happy to answer any questions or concerns. Thanks!

**Review by tswast — APPROVED** [review]
Thanks so much for the contribution!

File: tests/unit/test_compiler.py
Comment (tswast) [inline tests/unit/test_compiler.py:604]: This is failing lint.

```
./tests/unit/test_compiler.py:604:13: F541 f-string is missing placeholders
```

File: tests/unit/test_compiler.py
Comment (tswast) [inline tests/unit/test_compiler.py:475]: These are failing on ibis 2.x.

Would you mind creating a `tests/unit/test_compiler_geospatial.py` file with an importorskip and a minimum Ibis version? https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest-importorskip

**Review by tswast — CHANGES_REQUESTED** [review]

File: tests/unit/test_compiler.py
Comment (santind) [reply tests/unit/test_compiler.py]: Fixed with 2b09d5d

File: tests/unit/test_compiler.py
Comment (santind) [reply tests/unit/test_compiler.py:475]: It turns out that `ibis.geo_point()` is the historically-supported way of creating geo points, so I changed the unit test to use that.  I tested the change locally against Ibis 2.1.1 and it resolved the test failure.

**Comment by santind** [comment]
@tswast thanks for kicking off the unit tests.  Looks like the failure is related to some recent upstream dependency changes in ibis-framework around sqlalchemy, and you've fixed the problem in branch [`tswast-patch-1`](https://github.com/ibis-project/ibis-bigquery/tree/tswast-patch-1).

How would you like me to proceed? I can rebase to the `tswast-patch-1` branch, wait until it lands in `main` and rebase then, etc.

**Comment by cpcloud** [comment]
@santind #144 is almost ready to go, and ideally we can merge that soon to unblock this PR.

**Review by tswast — APPROVED** [review]
This is great. Thanks so much for the contribution!

**Comment by santind** [comment]
My pleasure!  Thanks for the feedback to get it over the line. 👍

**Pull request merged**
